core/src/test/java/io/questdb/std/datetime/millitime/DatesTest.java

```diff
@@ -127,6 +127,34 @@ public void testDayOfWeek() throws Exception {
         Assert.assertEquals(1, Dates.getDayOfWeekSundayFirst(millis));
     }
 
```

Comment:
It would be good to test max posssible values for day, week & week of month ?

Reply:
Willa dd some more tests here. I couldn't find any info on BC timestamps though, as none of the tests here seem to contain them - Only pre-unix tests, which i'll add. Also, I am not sure about the "correcT" definition of week of year/week of month. Strictly speaking, this calculation currently just uses days // 7 as the week; however it would probably be more reasonable to have this behave according to the "calendar" week - However, in this case we actually would need something like "sunday first" "monday first" for all of these calcualtions.

Reply:
Test have been extended with BC and corner cases now. Also, the logic seems to  be in-line with postgres to_char according to the documentation.

Reply:
👍